src/python.rs

```diff
@@ -127,57 +143,63 @@ impl PartialOrd<(u8, u8, u8)> for PythonVersionInfo<'_> {
 ///  * Thread 2 tries to locks the mutex, blocks
 ///  * Thread 1's Python interpreter call blocks trying to reacquire the GIL held by thread 2
 ///
-/// To avoid deadlocking, you should release the GIL before trying to lock a mutex, e.g. with
-/// [Python::allow_threads].
+/// To avoid deadlocking, you should release the GIL before trying to lock a mutex or `await`ing in
+/// asynchronous code, e.g. with [`Python::allow_threads`].
 ///
-/// # A note on Python reference counts
+/// # Releasing and freeing memory
 ///
-/// The [`Python`] type can be used to generate references to variables in
-/// Python's memory e.g. using [`Python::eval()`] and indirectly e.g.
-/// using [`PyModule::import()`], which takes a [`Python`] token
-/// as one if its arguments to prove the GIL is held.  The lifetime of these
-/// references is bound to the GIL (more precisely the [`GILPool`], see
-/// [`Python::new_pool()`]), which can cause surprising results with respect to
-/// when a variable's reference count is decreased so that it can be released to
-/// the Python garbage collector.  For example:
+/// The [`Python`] type can be used to create references to variables owned by the Python
```

Comment:
"owner by the Python interpreter" isn't quite true - the references are owned by PyO3 (but point to Python objects).

Reply:
I can see how the wording is confusing.

I'm (in general) not happy with how "references" are explained so far. Maybe it would be better to just use "owned (or strong) reference" and "borrowed (or weak) reference" everywhere, rather than "GIL-independent" and "GIL-dependent" or *just* "reference".

Reply:
Yes. Perhaps over the holiday season I will finally get another chance to look at #1056 and we can come up with better terminology then?